app/models/user.rb

```diff
@@ -200,6 +200,7 @@ def unread_inbox_comments_count
             not_forbidden_name: { if: :will_save_change_to_login? }
   validate :username_is_not_recently_changed, if: :will_save_change_to_login?
   validate :admin_username_generic, if: :will_save_change_to_login?
+  validate :username_must_be_different_from_current, on: :update, if: :username_change_attempt
```

Comment:
Could we use the built-in `will_save_change_to_login?` here, similar to the `username_is_not_recently_changed` validation?